lib/open_api_spex/open_api/map.ex

```diff
@@ -0,0 +1,395 @@
+
+  defp to_struct(%{type: "number"} = map, Schema) do
+    map
+    |> prepare_schema()
+    |> (&struct(Schema, &1)).()
+    |> to_struct(:xml, Xml)
+  end
+
+  defp to_struct(%{type: "integer"} = map, Schema) do
+    map
+    |> prepare_schema()
+    |> (&struct(Schema, &1)).()
+    |> to_struct(:xml, Xml)
+  end
+
+  defp to_struct(%{type: "boolean"} = map, Schema) do
+    map
+    |> prepare_schema()
+    |> (&struct(Schema, &1)).()
+    |> to_struct(:xml, Xml)
+  end
+
+  defp to_struct(%{type: "string"} = map, Schema) do
+    map
+    |> prepare_schema()
+    |> (&struct(Schema, &1)).()
+    |> to_struct(:xml, Xml)
+  end
+
+  defp to_struct(%{type: "array"} = map, Schema) do
+    map
+    |> prepare_schema()
+    |> (&struct(Schema, &1)).()
+    |> to_struct(:items, Schema)
+  end
+
+  defp to_struct(%{type: "object"} = map, Schema) do
+    map
+    |> prepare_schema()
+    |> (&struct(Schema, &1)).()
+    |> to_struct(:properties, Schemas)
+    |> to_struct(:externalDocs, ExternalDocumentation)
+  end
+
+  defp to_struct(%{anyOf: _valid_schemas} = map, Schema) do
+    Schema
+    |> struct(map)
+    |> to_struct(:anyOf, Schemas)
+    |> to_struct(:discriminator, Discriminator)
+  end
+
+  defp to_struct(%{oneOf: _valid_schemas} = map, Schema) do
+    Schema
+    |> struct(map)
+    |> to_struct(:oneOf, Schemas)
+    |> to_struct(:discriminator, Discriminator)
+  end
+
+  defp to_struct(%{allOf: _valid_schemas} = map, Schema) do
+    Schema
+    |> struct(map)
+    |> to_struct(:allOf, Schemas)
+    |> to_struct(:discriminator, Discriminator)
+  end
+
+  defp to_struct(%{not: _valid_schemas} = map, Schema) do
+    Schema
+    |> struct(map)
+    |> to_struct(:not, Schemas)
+  end
+
+  defp to_struct(map_or_list, Schemas), do: embedded_ref_or_struct(map_or_list, Schema)
+
+  defp to_struct(map, OAuthFlow) do
+    struct(OAuthFlow, map)
+  end
+
+  defp to_struct(map, Callback) do
+    map
+    |> Map.new(fn {k, v} ->
+      {k, to_struct(v, PathItem)}
+    end)
+  end
+
+  defp to_struct(map_or_list, Callbacks), do: embedded_ref_or_struct(map_or_list, Callback)
+
+  defp to_struct(map, Operation) do
+    Operation
+    |> struct(map)
+    |> to_struct(:tags, Tag)
+    |> to_struct(:externalDocs, ExternalDocumentation)
+    |> to_struct(:responses, Responses)
+    |> to_struct(:parameters, Parameters)
+    |> to_struct(:requestBody, RequestBody)
+    |> to_struct(:callbacks, Callbacks)
+    |> to_struct(:servers, Server)
+  end
+
+  defp to_struct(map, RequestBody) do
+    RequestBody
+    |> struct(map)
+    |> to_struct(:content, Content)
+  end
+
+  defp to_struct(map, RequestBodies), do: embedded_ref_or_struct(map, RequestBody)
+
+  defp to_struct(map, Parameter) do
+    map
+    # TODO: Must we do the below?
+    # |> convert_value_to_atom_if_present(:name)
+    |> convert_value_to_atom_if_present(:in)
+    |> convert_value_to_atom_if_present(:style)
+    |> (fn x -> struct(Parameter, x) end).()
```

Comment:
`(&struct(Parameter, &1).()` seems to be the pattern used above, maybe we could change this one as well to make it consistent with the others 😊

Reply:
These piped anonymous functions can all be replaced with `to_struct/2`. The clause on line 387 will handle it.

Reply:
@moxley to make that work you'd have to change the signature of the function that we're currently in though, right? Otherwise it'd just cause unwanted recursion.